src/test/java/seedu/address/logic/commands/GemCommandTest.java

```diff
@@ -0,0 +1,107 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static seedu.address.logic.commands.GemCommand.MESSAGE_SUCCESS;
+import static seedu.address.logic.commands.RequiredCommandTest.MISSING_MODULE_1;
+import static seedu.address.logic.commands.RequiredCommandTest.SINGLE_MODULE_PATH;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.exceptions.DataConversionException;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.ReadOnlyGradPad;
+import seedu.address.model.module.Module;
+import seedu.address.storage.GemCommandStorage;
+import seedu.address.storage.JsonGradPadStorage;
+
+class GemCommandTest {
+    public static final Path COMPILED_PATH_1 = Paths.get("src/test/data/GemCommandTest/compiledsem1GEmodules.json");
+    public static final Path COMPILED_PATH_2 = Paths.get("src/test/data/GemCommandTest/compiledsem2GEmodules.json");
+    public static final String TEST_GEH_SEM1_PATH = "src/main/resources/data/GEM/GEHs1.json";
+    public static final String TEST_GEH_SEM2_PATH = "src/main/resources/data/GEM/GEHsem2.json";
+    private Model model;
+    private GemCommand gemCommand = new GemCommand();
+    private ObservableList<Module> testModules;
+    public void setUpTestModules1() throws IOException, DataConversionException {
+        JsonGradPadStorage storage = new JsonGradPadStorage(Paths.get(TEST_GEH_SEM1_PATH));
+        ReadOnlyGradPad gradPad = storage.readGradPad().get();
+        testModules = gradPad.getModuleList();
+    }
+    public void setUpTestModules2() throws IOException, DataConversionException {
+        JsonGradPadStorage storage = new JsonGradPadStorage(Paths.get(TEST_GEH_SEM2_PATH));
+        ReadOnlyGradPad gradPad = storage.readGradPad().get();
+        testModules = gradPad.getModuleList();
+    }
+    public void setUpSingleModule() throws IOException, DataConversionException {
+        JsonGradPadStorage storage = new JsonGradPadStorage(SINGLE_MODULE_PATH);
+        ReadOnlyGradPad gradPad = storage.readGradPad().get();
+        testModules = gradPad.getModuleList();
+    }
+    public void setUpCompiledModules(Path path) throws IOException, DataConversionException {
+        JsonGradPadStorage storage = new JsonGradPadStorage(path);
+        ReadOnlyGradPad gradPad = storage.readGradPad().get();
+        testModules = gradPad.getModuleList();
+    }
```

Comment:
Since these 4 fxns have the same duplicate code, could the first 3 just make use of the 4th one? Then we can make the 4th fxn just a utility like `setUpModules(Path path)`
```suggestion
    public void setUpTestModules1() throws IOException, DataConversionException {
        setUpModules(TEST_GEH_SEM1_PATH));
    }
    public void setUpTestModules2() throws IOException, DataConversionException {
        setUpModules(Paths.get(TEST_GEH_SEM2_PATH));
    }
    public void setUpSingleModule() throws IOException, DataConversionException {
        setUpModules(SINGLE_MODULE_PATH);
    }
    public void setUpModules(Path path) throws IOException, DataConversionException {
        JsonGradPadStorage storage = new JsonGradPadStorage(path);
        ReadOnlyGradPad gradPad = storage.readGradPad().get();
        testModules = gradPad.getModuleList();
    }
```

Reply:
true! HAHAH i scraped all of them and made them into just a single setUpTestModules(Path path) that will just take in different paths when needed.